src/go/rpk/pkg/config/params.go

```diff
@@ -552,3 +568,133 @@ func (c *Config) addUnsetDefaults() {
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("%s: %s", p, value)
+			}
+			err = yaml.Unmarshal([]byte(in), i)
+			if err != nil {
+				return err
+			}
+			return nil
+		case "json":
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("{%q: %q}", p, value)
+			}
+			err = json.Unmarshal([]byte(in), i)
+			if err != nil {
+				return err
+			}
+			return nil
+		default:
+			return fmt.Errorf("unsupported format %s", format)
+		}
+	}
+	return errors.New("rpk bug, please describe how you encountered this at https://github.com/redpanda-data/redpanda/issues/new?assignees=&labels=kind%2Fbug&template=01_bug_report.md")
+}
+
+// getField deeply search in p for the value that reflect property props.
+func getField(props []string, p reflect.Value) (reflect.Value, reflect.Value, error) {
+	if len(props) == 0 {
+		return p, reflect.Value{}, nil
+	}
+	if p.Kind() == reflect.Slice && p.Len() > 0 {
+		p = p.Index(0)
+	}
+	if p.Kind() == reflect.Slice && p.Len() == 0 {
+		p.Set(reflect.New(p.Type().Elem()))
+	}
+	// If is a nil pointer we assign the zero value, and we reassign p to the
+	// value that p points to
+	if p.Kind() == reflect.Ptr {
+		if p.IsNil() {
+			p.Set(reflect.New(p.Type().Elem()))
+		}
+		p = reflect.Indirect(p)
+	}
+	if p.Kind() == reflect.Struct {
+		newP, other, err := getFieldByTag(props[0], p)
+		if err != nil {
+			return reflect.Value{}, reflect.Value{}, err
+		}
+		// if is "Other" map field, we stop the recursion and return
+		if (other != reflect.Value{}) {
+			// user may try to set deep unmanaged property:
+			// rpk.unmanaged.name = "name"
+			if len(props) > 1 {
+				return reflect.Value{}, reflect.Value{}, fmt.Errorf("cannot set property %v", strings.Join(props, "."))
+			}
+			return reflect.Value{}, other, nil
+		}
+		return getField(props[1:], newP)
+	}
+	return reflect.Value{}, reflect.Value{}, fmt.Errorf("cannot recurse on type %v", p.Type())
+}
+
+// getFieldByTag finds a field with a given yaml tag and returns 3 parameters:
+//
+//   1. if tag is found within the struct, return the field.
+//   2. if tag is not found _but_ the struct has "Other" field, return Other.
+//   3. Error if it can't find the given tag and "Other" field is unavailable.
+func getFieldByTag(tag string, p reflect.Value) (reflect.Value, reflect.Value, error) {
+	t := p.Type()
+	var other bool
+	// Loop struct to get the field that match tag.
+	for i := 0; i < p.NumField(); i++ {
+		// Rpk blindly set configuration parameters, we parse these parameters
+		// in the "Other" field.
+		if t.Field(i).Name == "Other" {
+			other = true
+		}
+		yt := t.Field(i).Tag.Get("yaml")
+
+		// yaml struct tags can contain flags such as omitempty,
+		// when tag.Get("yaml") is called it will return
+		//   "my_tag,omitempty"
+		// so we only need first parameter of the string slice.
+		ft := strings.Split(yt, ",")[0]
+
+		if ft == tag {
+			return p.Field(i), reflect.Value{}, nil
+		} else {
+			continue
+		}
```

Comment:
Unnecessary else case -- can just end the if. I'm surprised a linter doesn't catch this

Reply:
Yeah, I think we had something after that and we deleted it, can't remember. Thanks!